### ISSUE TITLE
SETUP FAILED on iOS

### ISSUE THREAD

**Issue by kaushiik** [j3k0/cordova-plugin-purchase#430]
I am trying to test in app purchase from my normal user but i got an error SETUP FAILED on app start. 

On app purchase i got an error Trying to purchase unknown product.

**Comment by AbuHani** [comment]
i have the same error, any updates ?

**Comment by dpa99c** [comment]
For those still encountering this, I have an explanation:

TL;DR: "Setup failed" means in-app purchases are not available on the iOS device.

If the iOS device has in-app purchases disabled (via Settings > Restrictions) or payment cannot be made for another reason (e.g. credit card register with Apple has expired), then the "Setup failed" message will be returned by the plugin.

The "Setup failed" message is [hard-coded into the JS layer of the plugin](https://github.com/j3k0/cordova-plugin-purchase/blob/v7.0.1/www/store-ios.js#L2139) and sent when the `setupFailed()` function is invoked by the native part of the plugin. This happens when [`canMakePayments` is FALSE](https://github.com/j3k0/cordova-plugin-purchase/blob/v7.0.1/src/ios/InAppPurchase.m#L285) which means the in-app purchases are not available for the above reasons. 

The native part of the plugin actually [returns a more useful error message: "Can't make payments"](https://github.com/j3k0/cordova-plugin-purchase/blob/v7.0.1/src/ios/InAppPurchase.m#L287) but the JS layer discards this and simply returns "Setup failed".

**Comment by anchann** [comment]
@dpa99c thank you for the helpful explanation.

Ideally, we should be able to display products available for purchase irrespective of the user's payment setup, and when they click on "buy", then the OS can tell them that their card expired, and guide them through the process required for turning `canMakePayments` to `true`.

Do you know if the current less than ideal behaviour is necessitated by the way storekit works, or is it just a function of how the plugin is implemented and can thus be improved?

In the interim, it sounds like displaying some text like "It seems that you have disabled in-app purchases in your settings, or something else about your setup is preventing access to in-app purchases" and pointing the user to the settings app is the best we can do.

**Comment by anchann** [comment]
@dpa99c Sorry, I should have just checked the iOS docs and answered the question myself. It does appear to be a storekit-level decision, not plugin level. The official word from the docs is:

```
An iPhone can be restricted from accessing the Apple App Store. For example, parents can restrict their children’s ability to purchase additional content. Your application should confirm that the user is allowed to authorize payments before adding a payment to the queue. Your application may also want to alter its behavior or appearance when the user is not allowed to authorize payments.
```

**Comment by anchann** [comment]
@dpa99c In the case of this failure, the plugin retries to initialize, indefinitely

```
function storekitInitFailed() {
    store.log.warn("ios -> storekit init failed");
    initializing = false;
    retry(storekitInit);
}
```

with the exponential backoff delay between retries capped at 120 seconds. For most users, this is a waste of CPU cycles. Is there a way to tell the plugin to stop trying?
